kvwrapper_etcd_v3/kvwrapper_etcd_v3.go

```diff
@@ -0,0 +1,180 @@
+package kvwrapper_etcd_v3
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	// "github.com/behance/go-common/kvwrapper"
+	"github.com/PinchasLev/go-common/kvwrapper"
+	log "github.com/behance/go-logging/log"
+	etcdv3 "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
+	// "google.golang.org/grpc"
+)
+
+// EtcdWrapper wraps the go-etcd client so it can implement the KVWrapper interface
+type EtcdV3Wrapper struct {
+	kapi etcdv3.KV
+	cli  etcdv3.Client
+}
+
+// NewKVWrapper returns a new kvwrapper_etcd as a KVWrapper
+func (e EtcdV3Wrapper) NewKVWrapper(servers []string, username, password string) kvwrapper.KVWrapper {
+	config := etcdv3.Config{
+		Endpoints:   servers,
+		Username:    username,
+		Password:    password,
+		DialTimeout: 5 * time.Second,
+	}
+	client, err := etcdv3.New(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return EtcdV3Wrapper{kapi: etcdv3.NewKV(client), cli: *client}
+}
+
+// Set sets the key = val with a ttl of ttl. If key is a path, it will be created.
+// The ttl will be handled via a lease
+// Leases are given in increments of seconds, and leases are granted in increments of seconds,
+// hence there is no need to convert to time.Duration (which is in nanoseconds) as is done in version 2
+// note! The API assunmes that a ttl of 0, implies a wish that the key/value pair not expire
+// implement taht in v3, by not acquiring a lease
+func (e EtcdV3Wrapper) Set(key string, val string, ttl uint64) error {
+	// acquire a lease to which we'll attach the new key/value pair
+	// lease.grant takes an int64 as the ttl, hence casting is necessary
+	log.Debug("called Set with key/value: ", key, val, " and ttl ", ttl)
+	if ttl == 0 {
```

Comment:
What happens if we are setting an existing key to a ttl of 0? This logic might ignore the new TTL causing unintended consequences. I don't know if we ever actually do this but might be worth calling out.

Reply:
Confirmed - although this is a curious pattern to keep interface compatibility, we do not use that functionality currently.